src/app/auth/services/auth.service.ts

```diff
@@ -15,10 +15,14 @@ export class Auth {
 
   constructor(private http: Http) {
     this.isLoggedIn = new Observable<boolean>(s => {
+      let prevState = undefined;
       Observable.timer(0, 1000).subscribe(x => {
         const token = sessionStorage.getItem('id_token');
         const isNotExpired = tokenNotExpired('id_token', token);
```

Comment:
`isNotExpired` never changes after it's declared and assigned an initial value. There is no statement like `isNotExpired = [whatever new value]`. It just gets emitted to the Observable:
```ts
s.next(isNotExpired);
```
In contrast to that, variable `prevState` changes, which is why it's declared with a keyword `let`.